rpm/src/test/java/org/eclipse/packager/rpm/signature/RpmFileSignatureProcessorTest.java

```diff
@@ -0,0 +1,113 @@
+package org.eclipse.packager.rpm.signature;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.bouncycastle.bcpg.ArmoredInputStream;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPPublicKey;
+import org.bouncycastle.openpgp.PGPPublicKeyRing;
+import org.bouncycastle.openpgp.bc.BcPGPPublicKeyRing;
+import org.eclipse.packager.rpm.RpmSignatureTag;
+import org.eclipse.packager.rpm.Rpms;
+import org.eclipse.packager.rpm.parse.InputHeader;
+import org.eclipse.packager.rpm.parse.RpmInputStream;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(OrderAnnotation.class)
+public class RpmFileSignatureProcessorTest {
+
+    private static final String RESULT_FILE_PATH = "src/test/resources/result/org.eclipse.scada-0.2.1-1.noarch.rpm";
+    private static final String RESULT_DIR = "src/test/resources/result";
+
+    @Test
+    @Order(1)
+    public void test_signing_existing_rpm() throws IOException, PGPException {
+        String passPhrase = "testkey";
+        File rpm = new File("src/test/resources/data/org.eclipse.scada-0.2.1-1.noarch.rpm");
+        File private_key = new File("src/test/resources/key/private_key.txt");
+        if (!rpm.exists() || !private_key.exists()) {
+            fail("Input files rpm or private_key does not exist");
+        }
+        InputStream rpmStream = new FileInputStream(rpm);
+        InputStream privateKeyStream = new FileInputStream(private_key);
+
+        RpmFileSignatureProcessor signatureProcessor = new RpmFileSignatureProcessor();
+        ByteArrayOutputStream signedPackage = signatureProcessor.perform(rpmStream, privateKeyStream, passPhrase);
+        byte[] bytes = signedPackage.toByteArray();
+
+        File resultDirectory = new File(RESULT_DIR);
+        resultDirectory.mkdir();
+        File signedRpm = new File(RESULT_FILE_PATH);
+        signedRpm.createNewFile();
+        FileOutputStream resultOut = new FileOutputStream(signedRpm);
+        resultOut.write(bytes);
+        resultOut.close();
+
+        RpmInputStream initialRpm = new RpmInputStream(new FileInputStream(rpm));
+        initialRpm.available();
```

Comment:
I have a hunch on what you are doing here :-) … but I think it makes sense explaining that in a brief sentence in the code.